src/SqlPersistence/SynchronizedStorage/StorageAdapter.cs

```diff
@@ -43,11 +44,21 @@ public Task<CompletableSynchronizedStorageSession> TryAdapt(OutboxTransaction tr
             return session;
         }
 
-        Transaction existingTransaction;
+        Transaction transportTx;
         // Transport supports DTC and uses TxScope owned by the transport
-        if (transportTransaction.TryGet(out existingTransaction))
+        transportTransaction.TryGet(out transportTx);
+        if (transportTx != null && Transaction.Current != null && !transportTx.Equals(Transaction.Current))
```

Comment:
I don't see `!=` defined in `Transaction` so I thought it would be safer to call `Equals` because equals is overridden and refers to *transaction hash*.